### PR TITLE
DTSPO-6455 Upgrade yarn to v3

### PR DESCRIPTION
**Pull request by JayDoubleu**
### JIRA link (if applicable) ###

[DTSPO-6455](https://tools.hmcts.net/jira/browse/DTSPO-6455)

### Change description ###

- Upgrade yarn to v3
- Enable yarn [pnp ](https://yarnpkg.com/features/pnp)
- Upgrade all node packages to latest

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,40 +2,133 @@
...
 pids
 *.pid
 *.seed
+*.pid.lock
 
 # Directory for instrumented libs generated by jscoverage/JSCover
 lib-cov
 
-# Coverage directory used by tools like nyc
+# Coverage directory used by tools like istanbul
```
Comment (timja) [inline .gitignore:22]: why change this? istanbul is deprecated? is this from a template?

Comment (JayDoubleu) [reply .gitignore:22]: Yes it is, stolen from : https://github.com/github/gitignore/blob/main/Node.gitignore

File: .yarnrc.yml [hunk 2]
```diff
@@ -0,0 +1 @@
+nodeLinker: pnp
```
Comment (timja) [inline .yarnrc.yml:1]: this is the default so not really required to add this file I believe.

File: Dockerfile [hunk 3]
```diff
@@ -1,10 +1,12 @@
 ARG PLATFORM=""
-FROM hmctspublic.azurecr.io/base/node${PLATFORM}:12-alpine as base
+FROM hmctspublic.azurecr.io/base/node${PLATFORM}:16-alpine as base
 
 USER root
 RUN apk add --no-cache python3 py3-pip make gcc g++
 COPY package.json yarn.lock ./
-RUN yarn install --production
+RUN corepack enable
+RUN yarn plugin import workspace-tools
```
Comment (timja) [inline Dockerfile:8]: could this be added into the `yarnrc` file so we don't need to install in docker each time?

File: Dockerfile [hunk 4]
```diff
@@ -1,10 +1,12 @@
 ARG PLATFORM=""
-FROM hmctspublic.azurecr.io/base/node${PLATFORM}:12-alpine as base
+FROM hmctspublic.azurecr.io/base/node${PLATFORM}:16-alpine as base
 
 USER root
 RUN apk add --no-cache python3 py3-pip make gcc g++
 COPY package.json yarn.lock ./
-RUN yarn install --production
+RUN corepack enable
```
Comment (timja) [inline Dockerfile:7]: could we collapse all these run statements into one line?

**Review by timja — APPROVED** [review]

**Pull request merged**